src/main/scala/com/snowplowanalytics/kinesistee/KinesisTee.scala

```diff
@@ -31,48 +32,39 @@ object KinesisTee extends Tee {
     * using the given routingStrategy. Transform the data using the given transformation strategy
     * first - and then use the given filter strategy to filter the transformed data
     * @param routingStrategy routing strategy to use
-    * @param transformationStrategy transformation strategy to use
-    * @param filterStrategy filtering strategy to use
+    * @param operationStrategy list sequence of operations to be applied
     * @param content list of records/content to tee on
     */
   def tee(routingStrategy: RoutingStrategy,
-          transformationStrategy: Option[TransformationStrategy],
-          filterStrategy: Option[FilterStrategy],
-          content: Seq[Content]): Unit = {
-
-    // transform first
-    // then filter
-    // then push to stream via StreamWriter
+          operationStrategy: List[Operator],
+          content: Seq[NonEmptyContent]) = {
 
-    def transform(content:Content) = {
-      transformationStrategy match  {
-        case Some(strategy) => {
-          strategy.transform(content) match {
-            case Success(s) => s
-            case Failure(f) => {
-              // see https://github.com/snowplow/kinesis-tee/issues/11
-              System.err.println(s"Error transforming item: '$content'\n\n:Reason: ${f.head.getMessage}")
-              content
-            }
-          }
-        }
-        case None => content
-      }
+    /**
+      * Apply operators in sequence
+      * @param content
+      * @return content
+      */
+    def operations(content: NonEmptyContent) = {
+      var transformedContent: ValidationNel[Throwable, Content] = NonEmptyContent(content.row, content.partitionKey).success
+      operationStrategy.foreach((operator: Operator) => {
+        transformedContent = operator(transformedContent)
+      })
+      transformedContent
     }
 
-    def filter(content:Content) = {
-      filterStrategy match {
-        case Some(strategy) => {
-          strategy.filter(content) match {
-            case Success(s) => s
-            case Failure(f) => {
-              // see https://github.com/snowplow/kinesis-tee/issues/11
-              System.err.println(s"Error filtering item '$content'\n\n:Reason: ${f.head.getMessage}")
-              false
-            }
-          }
-        }
-        case None => true
+    /**
+      * Filter FilteredContent and Failures
+      * @param content
+      * @return Boolean
+      */
+
```

Comment:
There's some floating scaladoc here too.

I'm not sure you can scaladoc inner functions actually? You can check the output of this with `sbt doc`